### PR TITLE
feat: Support more npm auth fields

### PR DESCRIPTION
**Pull request by tianfeng92**
# One-line summary

Support npm auth fields:
- `_auth`
- `username`
- `_password`
- `email`

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -136,18 +139,32 @@ export function getNpmConfig(runnerConfig: NpmConfigContainer) {
...
+        cfg[`${uri}_auth`] = r.auth;
+      }
+      if (r.username) {
+        cfg[`${uri}username`] = r.username;
+      }
+      if (r.password) {
+        cfg[`${uri}_password`] = r.password;
+      }
+      if (r.email) {
+        cfg[`${uri}email`] = r.email;
```
Comment (alexplischke) [inline src/utils.ts:167]: Missing `:`. See https://docs.npmjs.com/cli/v10/configuring-npm/npmrc#auth-related-configuration.

**Review by alexplischke — CHANGES_REQUESTED** [review]

File: src/utils.ts (hunk 1 above)
Comment (tianfeng92) [reply src/utils.ts:167]: The `:` is added in `getRegistryURI`, though it's a bit confusing. But I read the docs and it states 

> the scope may look like `//registry.npmjs.org/:`

So I put it into `getRegistryURI`. I can move it back.

Comment (alexplischke) [reply src/utils.ts:167]: Oh my bad, didn't see!
If you want to keep it in `getRegistryURI()`, then I'd recommend renaming that method to `getRegistryURIFragment`. Same with the local var `uri` -> `uriFragment`.
That way it's clearer that this is no mere URI.

Comment (alexplischke) [reply src/utils.ts:167]: >  Same with the local var uri -> uriFragment.

Still missing 👁️

**Review by alexplischke — CHANGES_REQUESTED** [review]

**Review by alexplischke — APPROVED** [review]

**Pull request merged**